datafusion/core/src/execution/context/mod.rs

```diff
@@ -495,6 +496,32 @@ impl SessionContext {
                 self.set_variable(stmt).await
             }
 
+            LogicalPlan::Statement(Statement::CreateFunction(stmt)) => {
+                let function_factory = self.state.read().function_factory.clone();
+
+                match function_factory {
+                    Some(f) => f.create(self.state.clone(), stmt).await?,
+                    None => Err(DataFusionError::Configuration(
+                        "Function factory has not been configured".into(),
+                    ))?,
+                };
```

Comment:
Minor: I think it would be better to follow the model of `create_view` etc and make this code a function on `SessionContext` (so that this code just dispatches to that method)

The upside of doing so is that it follows the existing code pattern and it offers a place to document `create_function` and `drop_function` publically in in more detail